src/CommunityVoices/App/Api/Controller/Tag.php

```diff
@@ -4,20 +4,50 @@
 
 use CommunityVoices\Model\Service\TagLookup;
 use CommunityVoices\Model\Service;
+use CommunityVoices\Model\Exception;
+
 use CommunityVoices\App\Api\Component;
 
 class Tag extends Component\Controller
 {
+    protected $recognitionAdapter;
     protected $tagLookup;
+    protected $tagManagement;
 
     public function __construct(
-        Service\TagLookup $tagLookup
+        Component\SecureContainer $secureContainer,
+        Component\RecognitionAdapter $recognitionAdapter,
+        Service\TagLookup $tagLookup,
+        Service\TagManagement $tagManagement
     ) {
+        parent::__construct($secureContainer);
+
+        $this->recognitionAdapter = $recognitionAdapter;
         $this->tagLookup = $tagLookup;
+        $this->tagManagement = $tagManagement;
     }
 
-    public function getAllTag()
+    protected function getAllTag()
     {
         $this->tagLookup->findAll();
     }
+    protected function getTagUpload()
+    {
+        // intentionally blank
+    }
+
+    protected function postTagUpload($request)
+    {
+        $identity = $this->recognitionAdapter->identify();
```

Comment:
Why do you have this line?  We should only get the identity if the management service needs it.

Reply:
Was removed